src/goto-checker/properties.h

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************\
+
+Module: Properties
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Properties
+
+#ifndef CPROVER_GOTO_CHECKER_PROPERTIES_H
+#define CPROVER_GOTO_CHECKER_PROPERTIES_H
+
+#include <unordered_map>
+
+#include <goto-programs/goto_model.h>
+
+/// The result status of a property
+enum class property_resultt
+{
+  /// The property was not reached (also used for initialization)
```

Comment:
Not reached yet, I guess -- what's the difference between this and UNKNOWN?

Reply:
It would rather be for unreachable assertions (for instance in methods that are loaded but not called from the entry point)

Reply:
That seems the same as `NOT_REACHABLE`. Perhaps they should be `NOT_REACHABLE` and `NOT_SATISFIABLE` (can get there, but can't concurrently satisfy the condition)?

Reply:
All add examples to clarify these notions.

Reply:
Assume we run a path-symex that does at most 5 unwindings and stops
before exploring the else branch (*).

```c
int main(int argc, char **argv)
{
  if(argc < 0)
    assert(0); // NOT_REACHABLE
  assert(argc >= 0); // PASS
  if(argc < 5)
  {
    for(int i=0; i<10; i++)
    {
      assert(i == 2) // FAIL
      if(i == 15)
        assert(0); // UNKNOWN
    }
  }
  else // (*)
  {
    assert(0); // NOT_REACHED
  }
}
```

Reply:
How about `NOT_REACHED` -> `NOT_CHECKED`? `NOT_REACHED` and `NOT_REACHABLE` are so close to synonyms it will surely confuse.